include/pinocchio/algorithm/jacobian.hxx

```diff
@@ -632,6 +632,12 @@ namespace pinocchio
       const Eigen::MatrixBase<Matrix6xLike> & J)
     {
       assert(model.check(data) && "data is not consistent with model.");
+      PINOCCHIO_CHECK_ARGUMENT_SIZE(
+        J.rows(), 6, "The numbers of rows of the Jacobian matrix is not equal to 6.");
+      PINOCCHIO_CHECK_ARGUMENT_SIZE(
+        J.cols(), model.nv,
+        "The numbers of columns in the Jacobian matrix does not math the "
+        "number of Dofs in the model.");
```

Comment:
Is it worth explicitly saying number of "velocity DOFS" so that users are prompted to use `nv` rather than `nq`?

It's a subtle detail that gets most people, and also "works" unless you have continuous, planar, or floating joints in your model.